pandas_dev_flaker/_plugins_tree/disallow_argument_types.py

```diff
@@ -0,0 +1,50 @@
+import ast
+from typing import Iterator, Tuple
+
+from pandas_dev_flaker._data_tree import State, register
+
+MSG = "PDF026 found union between Series and AnyArrayLike in type hint"
+
+
+@register(ast.FunctionDef)
+def visit_arg(
+    state: State,
+    node: ast.FunctionDef,
+    parent: ast.AST,
+) -> Iterator[Tuple[int, int, str]]:
+    arguments = node.args.args
+    for arg in arguments:
+        if isinstance(arg.annotation, ast.BinOp) and parse_annotation(
+            arg.annotation,
+        ):
+            yield arg.lineno, arg.col_offset, MSG
+
+
+def parse_annotation(node: ast.BinOp) -> bool:
+    series, any_array_like = "Series", "AnyArrayLike"
+    is_series, is_array_like = False, False
+
+    while True:
+        if isinstance(node.right, ast.Name):
+            if node.right.id == series:
+                is_series = True
+            if node.right.id == any_array_like:
+                is_array_like = True
+
+        if isinstance(node.left, ast.Name):
+            if node.left.id == series:
+                is_series = True
+            if node.left.id == any_array_like:
+                is_array_like = True
+
+        if isinstance(node.left, ast.BinOp):
+            node = node.left
+        elif isinstance(node.left, ast.Name) or isinstance(
+            node.left,
+            ast.Subscript,
+        ):
+            break
+        else:
+            raise AssertionError
```

Comment:
This is too specific, and can break with
```
    def __array__(self, dtype: npt.DTypeLike | None = None) -> np.ndarray:
```

Reply:
this could work?
`if not isinstance(node.left, ast.BinOp):
    break
else:
    node = node.left`

Reply:
might be fine